libs/types/src/locations.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2023 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::RuntimeDebugNoBound;
+use scale_info::TypeInfo;
+use sp_core::{H160, H256};
+use sp_runtime::{
+	traits::{BlakeTwo256, Hash},
+	AccountId32,
+};
+use xcm::{v1::MultiLocation, VersionedMultiLocation};
+
+use crate::domain_address::DomainAddress;
+
+/// Location types for destinations that can receive restricted transfers
+#[derive(Clone, RuntimeDebugNoBound, Encode, Decode, Eq, PartialEq, MaxEncodedLen, TypeInfo)]
+pub enum Location {
+	/// Local chain account sending destination.
+	Local(AccountId32),
+	/// Test
+	TestLocal(u64),
```

Comment:
Can we hide this behind a feature flag? and maybe have a fixed codec index of 255 in order to net mess up tests that rely on decoding?